src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+    Actor* thisx = &this->actor;
+
+    ActorCollider_AllocCylinder(globalCtx, &this->collider);
+    if (!thisx) {} // necessary to match
+    func_8005C450(globalCtx, &this->collider, thisx, &D_80AF0800);
+
+    ActorCollider_AllocCylinder(globalCtx, &this->collider2);
+    func_8005C450(globalCtx, &this->collider2, thisx, &D_80AF082C);
+}
+
+void func_80AEAD98(EnRu1* this, GlobalContext* globalCtx) {
+    ActorCollider_FreeCylinder(globalCtx, &this->collider);
+    ActorCollider_FreeCylinder(globalCtx, &this->collider2);
+}
+
+void func_80AEADD8(EnRu1* this) {
+    this->unk_34C = 0;
+}
+
+s32 func_80AEADE0(EnRu1* this) {
+    s32 params = this->actor.params >> 8;
+    return params & 0xFF;
+}
+
+s32 func_80AEADF0(EnRu1* this) {
+    s16 params = this->actor.params;
+    return params & 0xFF;
+}
+
+void EnRu1_Destroy(EnRu1* this, GlobalContext* globalCtx) {
+    func_80AEAD98(this, globalCtx);
+}
+
+void func_80AEAE1C(EnRu1* this) {
+    s32 pad[3];
+    s16* unk_25E = &this->unk_25E;
+    s16* unk_25C = &this->unk_25C;
+
+    if (!DECR(*unk_25E)) {
+        *unk_25E = Math_Rand_S16Offset(0x3C, 0x3C);
+    }
+
+    *unk_25C = *unk_25E;
+    if (*unk_25C >= 3) {
+        *unk_25C = 0;
+    }
+}
+
+void func_80AEAEA4(EnRu1* this, s16 arg1) {
+    this->unk_25C = arg1;
+}
+
+void func_80AEAEB8(EnRu1* this, s16 arg1) {
+    this->unk_260 = arg1;
+}
+
+void func_80AEAECC(EnRu1* this, GlobalContext* globalCtx) {
+    f32* velocityY = &this->actor.velocity.y;
+    f32 velocityYHeld = *velocityY;
+
+    *velocityY = -4.0f;
+    func_8002E4B4(globalCtx, &this->actor, 19.0f, 25.0f, 30.0f, 7);
+    *velocityY = velocityYHeld;
+}
+
+s32 func_80AEAF38(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        return 1;
+    }
+    return 0;
+}
+
+CsCmdActorAction* func_80AEAF58(GlobalContext* globalCtx, s32 actorActionIdx) {
+    s32 pad[2];
+    CsCmdActorAction* ret = NULL;
+
+    if (!func_80AEAF38(globalCtx)) {
+        ret = globalCtx->csCtx.actorActions[actorActionIdx];
+    }
+    return ret;
+}
+
+s32 func_80AEAFA0(GlobalContext* globalCtx, u16 action, s32 actorActionIdx) {
+    CsCmdActorAction* csCmdActorAction = func_80AEAF58(globalCtx, actorActionIdx);
+    if ((csCmdActorAction != NULL) && (csCmdActorAction->action == action)) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80AEAFE0(GlobalContext* globalCtx, u16 action, s32 actorActionIdx) {
+    CsCmdActorAction* csCmdActorAction = func_80AEAF58(globalCtx, actorActionIdx);
+    if ((csCmdActorAction != NULL) && (csCmdActorAction->action != action)) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80AEB020(EnRu1* this, GlobalContext* globalCtx) {
+    EnRu1* actorIt = globalCtx->actorCtx.actorList[ACTORTYPE_NPC].first;
+
+    while (actorIt != NULL) {
+        if (actorIt->actor.id == ACTOR_EN_RU1) {
+            if (actorIt != this) {
+                if (actorIt->action == 31 || actorIt->action == 32 || actorIt->action == 24) {
```

Comment:
I am happy combining the first two (and tested it to work fine), but the action check seems like a separate "kind" of check. I think while it may be possible to combine all of them, it's more of a style question. Are we combining ifs whenever possible? I'd argue there are some cases where they're best left as separate things for readability and such...but yeah, I'm open to whatever we all prefer